array.h

```diff
@@ -67,6 +68,12 @@
 #define NDARRAY_RESTRICT
 #endif
 
+#if defined(__CUDA__)
+#define NDARRAY_PRINT_ERR(...) printf(__VA_ARGS__)
+#else
+#define NDARRAY_PRINT_ERR(...) fprintf(stderr, __VA_ARGS__)
```

Comment:
The new assert code should probably just always use `fprintf(stderr, ...)`. That should avoid the need for `std::flush`, which shouldn't be needed anyways to see assert messages, and this might also fix the CUDA build.

Reply:
std::flush is unrelated to the new logging -  if a test crashed on an assertion failure, std::cout << test_name; won't be guaranteed to print the name of the test that failed (and it didn't in my repro).

I don't think fprintf(stderr) exists in cuda - that's why I have this #ifdef'd logic in the first place. https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#formatted-output

`make CXX=clang++ cuda_build_test` works on my machine :/ I need to check what's different on my machine vs. the build environment. If you get the chance, can you see if it works for you?